flowey/flowey_lib_common/src/install_apt_pkg.rs

```diff
@@ -170,10 +170,17 @@ impl FlowNode for Node {
                 let sh = xshell::Shell::new()?;
 
                 if !skip_update {
-                    xshell::cmd!(sh, "sudo apt-get update").run()?;
+                    // retry on failure in case another process has a lock on /var/lib/apt/lists/lock
+                    // DPkg::Lock::Timeout doesn't seem to work for apt-get update
+                    xshell::cmd!(sh, "bash -c 'i=0; while [ $i -lt 5 ] && ! sudo apt-get update; do let \"i=i+1\"; sleep 1; done;'").run()?;
```

Comment:
no need to do a bash loop here - you're already writing Rust, so just use a Rust for loop + `std::thread::sleep` here